.github/workflows/ci.yml

```diff
@@ -112,4 +112,33 @@ jobs:
                 build_py_project_in_conda_env
                 build_docs
 
+    dep_tests:
+        strategy:
+            matrix:
+                dep_project: [meshmode, grudge, pytential]
```

Comment:
Won't this shoot up the CI time by a lot. Right now loopy tests take ~30 minutes. After this PR, `pytential` would be the slowest CI with ~80 minutes. I would still prefer to upstream any bugs picked up by pytential into loopy's CI in an MWE fashion.

Another factor to account for is the presence of bugs in the depending project (like https://github.com/inducer/pytential/issues/57) that might slow down progress in this project.

Reply:
It will increase CI times, for sure. What's hurt by that though? Sure, I'd like our PR turn-around time to be 30 minutes or less, but we're *miles* from that point. A day or two, in an ideal scenario. Often much longer. For those times, an extra 80 minutes doesn't really hurt IMO.

> I would still prefer to upstream any bugs picked up by pytential into loopy's CI in an MWE fashion.

I don't agree. If loopy breaks a downstream project, then it is (IMO) immeasurably better to catch that *before* the merge into loopy than after. If it is caught after (as in the current misery with https://github.com/inducer/pytential/pull/56), development in pytential is effectively at a standstill until the loopy end of the problem is resolved. And there's at least three or four PRs that are already being held up by this.

FWIW, the endgame I envision for this PR is that it will show the pytential failure, we'll revert #197 and #217, merge, and everybody is off to the races again. We'll then retool those changes so as to not break pytential.

> Another factor to account for is the presence of bugs in the depending project (like inducer/pytential#57) that might slow down progress in this project.

Ultimately, all this does is gather additional information. We are as free to ignore the information this provides as we were before. So I kind of fail to see this as a negative.

Reply:
If you're worried about CI times, you can help by contributing a gitlab analog of these checks. On Gitlab, pytential CI takes a more modest 46-ish minutes.

Reply:
An additional aspect is that we're at 10 jobs now, which AFAIK is the limit that Github will run in parallel. So we're not yet at the point where these jobs are waiting for each other.

Reply:
Agreed. Sounds like from experience of maintaining the downstream the benefits of the PR outweigh its downsides. Thanks for the explanation!

Reply:
Thanks for understanding! These cross-project things mostly end up on my plate. They're painful to deal with, and they take up huge chunks of time.

Reply:
Would it be worth it to just `pytest.mark` some tests in the downstream projects to run here (and elsewhere)? Some that come to mind (that use most of the functionality)
* `test_scalar_int_eq.py` convergence tests in pytential
* the advection and Maxwell convergence tests in grudge

Reply:
:shrug: We could, depending on how painful it ends up being to live with these tests. For now, I think there's something to be said for covering most of the regular CI for those projects. Plus, I would like to get this in place quickly.